music_assistant/providers/audible/audible_helper.py

```diff
@@ -211,21 +370,37 @@ async def _call_api(self, path: str, **kwargs: Any) -> Any:
                 key=cache_key_with_params, base_key=CACHE_DOMAIN, category=CACHE_CATEGORY_API
             )
         if not response:
+            # Use the client that was passed in the constructor, which should be using
+            # the cached authenticator from the provider class
             response = await self.client.get(path, **kwargs)
             await self.mass.cache.set(
                 key=cache_key_with_params, base_key=CACHE_DOMAIN, data=response
             )
         return response
 
-    async def _parse_audiobook(self, audiobook_data: dict[str, Any]) -> Audiobook:
+    async def _parse_audiobook(self, audiobook_data: dict[str, Any] | None) -> Audiobook:
+        if audiobook_data is None:
+            self.logger.error("Received None audiobook_data in _parse_audiobook")
+            raise ValueError("Invalid audiobook data: received None")
+
```

Comment:
See my comments above - you are already checking for None - this parse method should only parse a Audible object to a MA Audiobook